### DISCUSSION TITLE
Better SEO options in edit articles

### DISCUSSION THREAD

**Discussion by JoomlaFan** [joomla/joomla-cms#15833] [2. Cold Features]
I think this may improve the SEO option, grouped in one tab.

![screen shot 2017-05-05 at 11 20 34](https://issues.joomla.org/uploads/1/ba939b87f14827f9a64bf859ebc9b87c.png)

**Comment by tomekomel** [comment]
Nice improvement!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15833">issues.joomla.org/tracker/joomla-cms/15833</a>.</sub>

**Comment by brianteeman** [comment]
I dont personally see the benefit of moving these options from the publishing tab to a dedicated tab - but if you do then feel free to submit a PR to make this change. Note that for consistency you would need to make the same change in all the core components. Also you should ensure that the option to hide the publishing tab still works etc

**Comment by dgrammatiko** [comment]
8 tabs that someone needs to go through to publish an article? Really?
We are doing it wrong!

**Comment by ghost** [comment]
Why ist this when or where "SEO"???

- Author
- Content Rights
- Browser Page Title
...

I think this should not go into core. It's easy to handle things like this with a short custom plugin.

**Comment by brianteeman** [comment]
@bertmert this is just moving items from the publishing tab to their own tab

**Comment by ghost** [comment]
@brianteeman 

> this is just moving items from the publishing tab to their own tab

Yes, I know. But these options are not all "SEO" or not only "SEO" or maybe sometimes "SEO" or ... The tabulator label is misleading nonsense.

**Comment by JoomlaFan** [comment]
These options are now scattered. For example, see where in Joomla 3.7 was added override the article title meta - in the article parameters tab - at the bottom. It is really difficult to find in the article options. If this is not improved, the opinion about SEO in Joomla will still be bad.

Meta title, description, keywords and robots should be together. Not necessarily on one card. But a separate tab is in my opinion the best one.

Sorry for my english ;-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15833">issues.joomla.org/tracker/joomla-cms/15833</a>.</sub>

**Comment by photodude** [comment]
I'm not for moving it to a new tab, but not against grouping as an "SEO" section  under the existing publishing tab.

**Comment by Quy** [comment]
Similar to #12043

**Comment by brianteeman** [comment]
I know that this is really just about moving things around but at this point in Joomla  3 lifespan I am not in favour of what will actually be quite a disruptive change to peoples workflow and documentation etc.

I am going to set it to be reevaluated for Joomla 4 where such a change would be more appropriate.

**Comment by uglyeoin** [comment]
I think having it an SEO tab changes the perception of it.  We all know (probably) that SEO is quite detailed work and there's no silver bullet.  But many end users of Joomla! think that Yoast for WordPress is a silver bullet.  And therefore Joomla! is missing this.  Which is in my opinion not correct but it's not about correctness it's about perception and  ensuring users find Joomla! easy to use and has good UX.  

I very much like the idea of the visual representation in #12043 and I also like the count which helps users to understand how to be SEO friendly.  I think a lot of people will see page title for this usage and I think 3rd party extensions should offer much more than this as functionality (and some do).  And those additional features do not belong in the core (IMHO).

Also, I would like to see these fields filled in automatically from the article for a new single article.    Or from the category from a category blog view.  If this could be standardised across Joomla! that would be lovely.   By that I mean making it an API or function (or whatever the right word is) and publishing this information so that component makers can also use this feature for their components thus saving everyone time and making them more SEO friendly.  I also believe Joomla! looks at the menu first and if it is empty goes to the article.  By having this data in the first place is a micro performance improvement (I'm guessing)

**Comment by uglyeoin** [comment]
Additional fields that could be added are the social open graph meta tags http://ogp.me/ which are pretty standard fields now-a-days and should not require a plugin

**Comment by uglyeoin** [comment]
From the other issue.  
![68747470733a2f2f6973737565732e6a6f6f6d6c612e6f72672f75706c6f6164732f312f63303430656636363439353738623532323866646261613738633463383439642e6a7067](https://user-images.githubusercontent.com/4665225/49532974-aec95800-f8b5-11e8-81cf-66c9432ed501.jpg)

**Comment by anibalsanchez** [comment]
I think that we can't add more tabs and fields in the backend.

If 7 is a magic number to deal with complexity in UX, we are way beyond that limit. 

BTW: Remember to test this on a smartphone

**Comment by uglyeoin** [comment]
Well yes, of course, and I think there are tonnes of options, but in other systems there are these options too.  Unfortunately we cannot change the backend layout.  Other systems just don't stick them all in tabs in a row.  The possible perception of Joomla! being complicated, with bad UX and hard to use could/should be resolved in J4.  I think all of this stuff has to be a Joomla! 4 issue anyway because it could break B/C otherwise.  And in J4 we DO have the opportunity to change the admin layout.  But we can therefore add this SEO idea too as that will make a lot of people switch on to the fact this is possible.

**Comment by anibalsanchez** [comment]
There is always a good reason to add a new field, tab, or checkbox. I'm not questioning the need for improving the SEO perception, or how it can be achieved.

My point is that we always find the same action _adding one more item to a long list of fields, tabs, or checkboxes_.

**Comment by uglyeoin** [comment]
Yes, and unfortunately, without breaking b/c it's hard to change this action.  But perhaps it is something to consider for J4.  Although it's a little late perhaps.  But the idea that we could add an infinite amount of items (unrealistic but a concept) without breaking everything should be at least discussed.  But not here.  Here I just want to back up this person and the other person who had the same idea because I feel that although there may be many other good ideas, the feedback I get from customers is that they want to see this.

**Comment by AndySDH** [comment]
I'm all for this. This makes a lot of sense to group all the SEO-related options together. Right now finding the option for the "Browser Page Title" aka Meta Title is almost impossible, buried at the end of the Options tab. No reason for it to not be together with Meta Description etc.

> 8 tabs that someone needs to go through to publish an article? Really?
> We are doing it wrong!

Totally disagree. Nobody _needs_ to go through any tab other than the "Content" tab (and maybe the "Images" tab). All the rest is optional stuff that is OFFERED in addition to the basic stuff that you can do in the first tab.

So you'd propose stripping away functionallity or jamming it all in one page instead, making it a lot more confusing?

Joomla is very powerful and has a lot of additional features IN ADDITION to the simple stuff. Nothing wrong with that.

**Comment by wilsonge** [comment]
> I think having it an SEO tab changes the perception of it. We all know (probably) that SEO is quite detailed work and there's no silver bullet. But many end users of Joomla! think that Yoast for WordPress is a silver bullet. And therefore Joomla! is missing this. Which is in my opinion not correct but it's not about correctness it's about perception and ensuring users find Joomla! easy to use and has good UX.

I 100% agree with @uglyeoin on this. This was intended for this years GSOC project (unfortunately we couldn't get a mentor team on this). But it's definitely a valid feature request

**Comment by brianteeman** [comment]
I have a PR to submit in the morning to kill the useless meta keywords

**Comment by uglyeoin** [comment]
@brianteeman doesn't related articles use that?

I think Yahoo also uses it as a very minor signal. I'd be careful not to kill someone's Yahoo #1 just in case. But it's more of a FYI than a major counter argument I'm putting up.

**Comment by ghost** [comment]
> doesn't related articles use that

Module Articles Related use Meta Keywords.

**Comment by AndySDH** [comment]
@wilsonge any word on this since you agreed with the idea?

**Comment by brianteeman** [comment]
As with every "idea" it needs to scratch someone's itch enough for them to write the code.

**Comment by uglyeoin** [comment]
For me I wouldn't scratch the itch unless someone tells me it would be accepted. And for that I would need to fully understand the process for it being accepted. Like who makes the final decision. This idea has been raised several times and yet this issue is closed. So I wouldn't write the code purely for that reason at the moment. 

Since it's been raised a fee times I'm sure I'm not the only one who would scratch the itch. Someone will have to tell the naysayers that it's being accepted and green light the idea

**Comment by AndySDH** [comment]
I think at the very LEAST, we should move the Browser Page Title above the Meta Description field (in the Publishing tab). That would be a start and an improvement already. I could do the code for that, it's a 2 line fix.

**Comment by AndySDH** [comment]
Actually not so easy as I have no idea where the code for that is. The XML looks like a mess with everything out of order...

**Comment by wilsonge** [comment]
> This idea has been raised several times and yet this issue is closed

The issue never got closed... Just no one worked on it. I'm still happy to have an SEO tab if someone wants to work on it. But at this point would need to be to the 4.1 branch

**Comment by uglyeoin** [comment]
Ah yes, the other issues got closed.  Ok, I will try to find some time one day to work on this on the condition that it will defintiely get put through.  If there is going to be a debate at the END after I've done the work then please get the powers that be to either say that debate is null and void or else have that debate up front, I don't want to work on something if it's not going to be accepted I don't have that level of spare time at the moment.

**Comment by uglyeoin** [comment]
Happy to team up @AndySDH if you want to work together

**Comment by AndySDH** [comment]
I'm up for it, just haven't found the file/code where this is the controlled :D

Also, another idea:

If we add a new SEO Tab, we could streamline the tabs a little bit more, by moving the "Configure Edit Screen" options under a block inside the "Options" tab instead of being a separate tab. I don't think it deserves its own tab (anymore).

**Comment by uglyeoin** [comment]
I think perhaps that would be two pull requests.  And there is information in other places about where to find the stuff.  I think it comes out as JSON or something somewhere.  It's not so straight forward as moving things in an XML file is what I can remember.  Send me a message via www.squareballoon.co.uk and we'll work out how to do a fix.  BUT... only if it's going to get approved.

**Comment by uglyeoin** [comment]
Does anyone know if this would be approved?  Is there a visible, transparent approval process anywhere?  I have no interest in investing any time into things if they won't be accepted, but if told up front that they will then there's a chance I may work on these things.  But it has never been clear to me how this works, how to approach this process and therefore, how to volunteer for Joomla.

**Comment by brianteeman** [comment]
I GUESS that you should now use the RFC process to satisfy your question. It is explained in the latest issue of the magazine. It would not be my suggested workflow but it appears to be the official one now.

**Comment by uglyeoin** [comment]
Thanks @brianteeman that's really useful.  And I'm glad there is a defined process.  I can't think of a better or more public way to do things but you have more knowledge than I on that side of things.  Here's the article in case anyone else wants to read it.  https://magazine.joomla.org/all-issues/december/a-new-proposal-process

@joomlafan can you change your title to include RFC at the beginning of it?  Once done I think it's already had a lot of the discussion on it so it might not be a very long first phase.  Who knows.

**Reply by ghost** [reply]
thanks for providing the link to the article.
